src/datasource.ts

```diff
@@ -92,9 +92,10 @@ export class GenericDatasource {
     });
   }
 
-  metricFindQuery(query) {
+  findMetricsQuery(type: string, query: string) {
     const interpolated = {
-      target: this.templateSrv.replace(query, null, 'regex'),
+      type,
+      target: this.templateSrv.replace(query || '', null, 'regex'),
```

Comment:
Why `query || ''`? I think we can safely use just `query` if it's typed as `string`

Reply:
This comes from the angular $query- I have no idea if it can be null and if that would cause any runtime issue?

Reply:
We don't handle it now, I wouldn't chnage it until it cases issues.

Reply:
Should be handled there https://github.com/simPod/grafana-json-datasource/pull/55/files#r317408822. Here we expect `string`

Reply:
Cannot access the link anymore. Please feel free to change as you need.

Reply:
Just drop the `|| ''` here and leave it in `query_ctrl`